eisbuk-admin/src/react-redux-firebase/reducer/index.ts

```diff
@@ -0,0 +1,83 @@
+import { Action } from "@/enums/store";
+
+import {
+  LocalStore,
+  FirestoreAction,
+  FirestoreReducerAction,
+  FirestoreState,
+} from "@/types/store";
+
+const defaultState: LocalStore["firestore"] = {
+  data: {},
+  listeners: {},
+};
+
+export default (
+  state: FirestoreState = defaultState,
+  action: FirestoreReducerAction<FirestoreAction>
+): FirestoreState => {
+  switch (action.type) {
+    case Action.UpdateFirestoreListener:
+      const { collection, listener } =
+        action.payload as FirestoreReducerAction<Action.UpdateFirestoreListener>["payload"];
+      return {
+        ...state,
+        listeners: {
+          ...state.listeners,
+          [collection]: { ...state.listeners[collection], ...listener },
+        },
+      };
+
+    case Action.DeleteFirestoreListener:
+      const collectionToDelete =
+        action.payload as FirestoreReducerAction<Action.DeleteFirestoreListener>["payload"];
+      // remove the data from the subscribed collection from the store
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      const { [collectionToDelete]: _coll, ...data } = state.data;
+      // remove the listener for a collection from the store
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      const { [collectionToDelete]: _listener, ...listeners } = state.listeners;
+      // return state with listener for collection and the correspoding data removed
+      return { data, listeners };
+
+    case Action.UpdateLocalDocuments:
+      const { collection: collectionToUpdate1, data: updatedCollection1 } =
```

Comment:
I believe it would be better to use the variable name `collectionToUpdate` enclosing the whole code for this `case` in a block.